service/geopmdpy/varrun.py

```diff
@@ -571,11 +533,110 @@ def _load_session_file(self, sess_path):
             if batch_pid is not None and \
                self._is_pid_valid(batch_pid, file_time) == False:
                 sess.pop('batch_server')
+            self._sessions[client_pid] = dict(sess)
+            self._update_session_file(client_pid)
         else:
-            # Invalid session, remove batch server
-            sess['client_pid'] = self._INVALID_PID
-            if batch_pid is not None:
-                sess.pop('batch_server')
+           os.rename(sess_path, renamed_path)
+
+
+class WriteLock(object):
+    def __init__(self, var_path=GEOPM_SERVICE_VAR_PATH):
+        self._VAR_PATH = var_path
+        self._LOCK_PATH = os.path.join(self._VAR_PATH, "CONTROL_LOCK")
+        # Create an empty lock file if it does not exist
+        lock_pid = self.try_lock()
```

Comment:
assumes that /var/run/geopm-service exists, probably should call the secure_make_dirs() method here.